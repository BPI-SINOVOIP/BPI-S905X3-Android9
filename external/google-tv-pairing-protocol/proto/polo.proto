// Copyright 2009 Google Inc. All Rights Reserved.

package polo.wire.protobuf;

//removing deprecated option.
//refer go/namespacing-nano for additional details.
//option javanano_use_deprecated_package = true;
option java_outer_classname = "PoloProto";
option java_package = "com.google.polo.wire.protobuf";
option optimize_for = LITE_RUNTIME;

// OuterMessage - base outer message type used in the protocol.

message OuterMessage {

  // MessageType indicates the type of the enclosed message (serialized in the
  // `payload` field)
  enum MessageType {
    // Initialization phase
    MESSAGE_TYPE_PAIRING_REQUEST = 10;
    MESSAGE_TYPE_PAIRING_REQUEST_ACK = 11;

    // Configuration phase
    MESSAGE_TYPE_OPTIONS = 20;
    MESSAGE_TYPE_CONFIGURATION = 30;
    MESSAGE_TYPE_CONFIGURATION_ACK = 31;

    // Pairing phase
    MESSAGE_TYPE_SECRET = 40;
    MESSAGE_TYPE_SECRET_ACK = 41;
  }

  // Protocol status states.
  enum Status {
    STATUS_OK = 200;
    STATUS_ERROR = 400;
    STATUS_BAD_CONFIGURATION = 401;
    STATUS_BAD_SECRET = 402;
  }

  required uint32 protocol_version = 1 [default = 1];

  // Protocol status. Any status other than STATUS_OK implies a fault.
  required Status status = 2;

  // Encapsulated message.  These fields are required if status is STATUS_OK.
  optional MessageType type = 3;
  optional bytes payload = 4;

}


//
// Initialization messages
//

message PairingRequest {
  // String name of the service to pair with.  The name used should be an
  // established convention of the application protocol.
  required string service_name = 1;

  // Descriptive name of the client.
  optional string client_name = 2;
}

message PairingRequestAck {
  // Descriptive name of the server.
  optional string server_name = 1;
}


//
// Configuration messages
//

message Options {
  message Encoding {
    enum EncodingType {
      ENCODING_TYPE_UNKNOWN = 0;
      ENCODING_TYPE_ALPHANUMERIC = 1;
      ENCODING_TYPE_NUMERIC = 2;
      ENCODING_TYPE_HEXADECIMAL = 3;
      ENCODING_TYPE_QRCODE = 4;
    }

    required EncodingType type = 1;
    required uint32 symbol_length = 2;
  }

  enum RoleType {
    ROLE_TYPE_UNKNOWN = 0;
    ROLE_TYPE_INPUT = 1;
    ROLE_TYPE_OUTPUT = 2;
  }

  // List of encodings this endpoint accepts when serving as an input device.
  repeated Encoding input_encodings = 1;

  // List of encodings this endpoint can generate as an output device.
  repeated Encoding output_encodings = 2;

  // Preferred role, if any.
  optional RoleType preferred_role = 3;
}

message Configuration {
  // The encoding to be used in this session.
  required Options.Encoding encoding = 1;

  // The role of the client (ie, the one initiating pairing). This implies the
  // peer (server) acts as the complementary role.
  required Options.RoleType client_role = 2;
}

message ConfigurationAck {
}


//
// Pairing messages
//

message Secret {
  required bytes secret = 1;
}

message SecretAck {
  required bytes secret = 1;
}

