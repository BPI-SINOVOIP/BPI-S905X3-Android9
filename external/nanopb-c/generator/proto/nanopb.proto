// Custom options for defining:
// - Maximum size of string/bytes
// - Maximum number of elements in array
//
// These are used by nanopb to generate statically allocable structures
// for memory-limited environments.

import "google/protobuf/descriptor.proto";

option java_package = "fi.kapsi.koti.jpa.nanopb";

enum FieldType {
    FT_DEFAULT = 0; // Automatically decide field type, generate static field if possible.
    FT_CALLBACK = 1; // Always generate a callback field.
    FT_POINTER = 4; // Always generate a dynamically allocated field.
    FT_STATIC = 2; // Generate a static field or raise an exception if not possible.
    FT_IGNORE = 3; // Ignore the field completely.
}

// This is the inner options message, which basically defines options for
// a field. When it is used in message or file scope, it applies to all
// fields.
message NanoPBOptions {
  // Allocated size for 'bytes' and 'string' fields.
  optional int32 max_size = 1;
  
  // Allocated number of entries in arrays ('repeated' fields)
  optional int32 max_count = 2;
  
  // Force type of field (callback or static allocation)
  optional FieldType type = 3 [default = FT_DEFAULT];
  
  // Use long names for enums, i.e. EnumName_EnumValue.
  optional bool long_names = 4 [default = true];
  
  // Add 'packed' attribute to generated structs.
  // Note: this cannot be used on CPUs that break on unaligned
  // accesses to variables.
  optional bool packed_struct = 5 [default = false];
}

// Extensions to protoc 'Descriptor' type in order to define options
// inside a .proto file.
//
// Protocol Buffers extension number registry
// --------------------------------
// Project:  Nanopb
// Contact:  Petteri Aimonen <jpa@kapsi.fi>
// Web site: http://kapsi.fi/~jpa/nanopb
// Extensions: 1010 (all types)
// --------------------------------

extend google.protobuf.FileOptions {
    optional NanoPBOptions nanopb_fileopt = 1010;
}

extend google.protobuf.MessageOptions {
    optional NanoPBOptions nanopb_msgopt = 1010;
}

extend google.protobuf.EnumOptions {
    optional NanoPBOptions nanopb_enumopt = 1010;
}

extend google.protobuf.FieldOptions {
    optional NanoPBOptions nanopb = 1010;
}


