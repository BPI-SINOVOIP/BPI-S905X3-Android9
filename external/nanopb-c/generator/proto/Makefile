all: nanopb_pb2.py plugin_pb2.py

%_pb2.py: %.proto
	aprotoc --python_out=. $<
