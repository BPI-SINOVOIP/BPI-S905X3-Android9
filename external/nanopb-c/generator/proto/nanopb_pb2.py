# Generated by the protocol buffer compiler.  DO NOT EDIT!

from google.protobuf import descriptor
from google.protobuf import message
from google.protobuf import reflection
from google.protobuf import descriptor_pb2
# @@protoc_insertion_point(imports)


DESCRIPTOR = descriptor.FileDescriptor(
  name='nanopb.proto',
  package='',
  serialized_pb='\n\x0cnanopb.proto\x1a google/protobuf/descriptor.proto\"\x92\x01\n\rNanoPBOptions\x12\x10\n\x08max_size\x18\x01 \x01(\x05\x12\x11\n\tmax_count\x18\x02 \x01(\x05\x12$\n\x04type\x18\x03 \x01(\x0e\x32\n.FieldType:\nFT_DEFAULT\x12\x18\n\nlong_names\x18\x04 \x01(\x08:\x04true\x12\x1c\n\rpacked_struct\x18\x05 \x01(\x08:\x05\x66\x61lse*Z\n\tFieldType\x12\x0e\n\nFT_DEFAULT\x10\x00\x12\x0f\n\x0b\x46T_CALLBACK\x10\x01\x12\x0e\n\nFT_POINTER\x10\x04\x12\r\n\tFT_STATIC\x10\x02\x12\r\n\tFT_IGNORE\x10\x03:E\n\x0enanopb_fileopt\x12\x1c.google.protobuf.FileOptions\x18\xf2\x07 \x01(\x0b\x32\x0e.NanoPBOptions:G\n\rnanopb_msgopt\x12\x1f.google.protobuf.MessageOptions\x18\xf2\x07 \x01(\x0b\x32\x0e.NanoPBOptions:E\n\x0enanopb_enumopt\x12\x1c.google.protobuf.EnumOptions\x18\xf2\x07 \x01(\x0b\x32\x0e.NanoPBOptions:>\n\x06nanopb\x12\x1d.google.protobuf.FieldOptions\x18\xf2\x07 \x01(\x0b\x32\x0e.NanoPBOptionsB\x1a\n\x18\x66i.kapsi.koti.jpa.nanopb')

_FIELDTYPE = descriptor.EnumDescriptor(
  name='FieldType',
  full_name='FieldType',
  filename=None,
  file=DESCRIPTOR,
  values=[
    descriptor.EnumValueDescriptor(
      name='FT_DEFAULT', index=0, number=0,
      options=None,
      type=None),
    descriptor.EnumValueDescriptor(
      name='FT_CALLBACK', index=1, number=1,
      options=None,
      type=None),
    descriptor.EnumValueDescriptor(
      name='FT_POINTER', index=2, number=4,
      options=None,
      type=None),
    descriptor.EnumValueDescriptor(
      name='FT_STATIC', index=3, number=2,
      options=None,
      type=None),
    descriptor.EnumValueDescriptor(
      name='FT_IGNORE', index=4, number=3,
      options=None,
      type=None),
  ],
  containing_type=None,
  options=None,
  serialized_start=199,
  serialized_end=289,
)


FT_DEFAULT = 0
FT_CALLBACK = 1
FT_POINTER = 4
FT_STATIC = 2
FT_IGNORE = 3

NANOPB_FILEOPT_FIELD_NUMBER = 1010
nanopb_fileopt = descriptor.FieldDescriptor(
  name='nanopb_fileopt', full_name='nanopb_fileopt', index=0,
  number=1010, type=11, cpp_type=10, label=1,
  has_default_value=False, default_value=None,
  message_type=None, enum_type=None, containing_type=None,
  is_extension=True, extension_scope=None,
  options=None)
NANOPB_MSGOPT_FIELD_NUMBER = 1010
nanopb_msgopt = descriptor.FieldDescriptor(
  name='nanopb_msgopt', full_name='nanopb_msgopt', index=1,
  number=1010, type=11, cpp_type=10, label=1,
  has_default_value=False, default_value=None,
  message_type=None, enum_type=None, containing_type=None,
  is_extension=True, extension_scope=None,
  options=None)
NANOPB_ENUMOPT_FIELD_NUMBER = 1010
nanopb_enumopt = descriptor.FieldDescriptor(
  name='nanopb_enumopt', full_name='nanopb_enumopt', index=2,
  number=1010, type=11, cpp_type=10, label=1,
  has_default_value=False, default_value=None,
  message_type=None, enum_type=None, containing_type=None,
  is_extension=True, extension_scope=None,
  options=None)
NANOPB_FIELD_NUMBER = 1010
nanopb = descriptor.FieldDescriptor(
  name='nanopb', full_name='nanopb', index=3,
  number=1010, type=11, cpp_type=10, label=1,
  has_default_value=False, default_value=None,
  message_type=None, enum_type=None, containing_type=None,
  is_extension=True, extension_scope=None,
  options=None)


_NANOPBOPTIONS = descriptor.Descriptor(
  name='NanoPBOptions',
  full_name='NanoPBOptions',
  filename=None,
  file=DESCRIPTOR,
  containing_type=None,
  fields=[
    descriptor.FieldDescriptor(
      name='max_size', full_name='NanoPBOptions.max_size', index=0,
      number=1, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    descriptor.FieldDescriptor(
      name='max_count', full_name='NanoPBOptions.max_count', index=1,
      number=2, type=5, cpp_type=1, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    descriptor.FieldDescriptor(
      name='type', full_name='NanoPBOptions.type', index=2,
      number=3, type=14, cpp_type=8, label=1,
      has_default_value=True, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    descriptor.FieldDescriptor(
      name='long_names', full_name='NanoPBOptions.long_names', index=3,
      number=4, type=8, cpp_type=7, label=1,
      has_default_value=True, default_value=True,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    descriptor.FieldDescriptor(
      name='packed_struct', full_name='NanoPBOptions.packed_struct', index=4,
      number=5, type=8, cpp_type=7, label=1,
      has_default_value=True, default_value=False,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
  ],
  extensions=[
  ],
  nested_types=[],
  enum_types=[
  ],
  options=None,
  is_extendable=False,
  extension_ranges=[],
  serialized_start=51,
  serialized_end=197,
)

import google.protobuf.descriptor_pb2

_NANOPBOPTIONS.fields_by_name['type'].enum_type = _FIELDTYPE

class NanoPBOptions(message.Message):
  __metaclass__ = reflection.GeneratedProtocolMessageType
  DESCRIPTOR = _NANOPBOPTIONS
  
  # @@protoc_insertion_point(class_scope:NanoPBOptions)

nanopb_fileopt.message_type = _NANOPBOPTIONS
google.protobuf.descriptor_pb2.FileOptions.RegisterExtension(nanopb_fileopt)
nanopb_msgopt.message_type = _NANOPBOPTIONS
google.protobuf.descriptor_pb2.MessageOptions.RegisterExtension(nanopb_msgopt)
nanopb_enumopt.message_type = _NANOPBOPTIONS
google.protobuf.descriptor_pb2.EnumOptions.RegisterExtension(nanopb_enumopt)
nanopb.message_type = _NANOPBOPTIONS
google.protobuf.descriptor_pb2.FieldOptions.RegisterExtension(nanopb)
# @@protoc_insertion_point(module_scope)
