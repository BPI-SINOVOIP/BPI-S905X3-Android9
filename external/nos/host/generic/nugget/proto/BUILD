################################################################################
# proto cc libraries
################################################################################

cc_proto_library(
    name = "nugget_app_keymaster_keymaster_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_keymaster_keymaster_proto",
    ],
)

cc_proto_library(
    name = "nugget_app_protoapi_control_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_protoapi_control_proto",
    ],
)

cc_proto_library(
    name = "nugget_app_protoapi_diagnostics_api_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_protoapi_diagnostics_api_proto",
    ],
)

cc_proto_library(
    name = "nugget_app_protoapi_testing_api_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_protoapi_testing_api_proto",
    ],
)

cc_proto_library(
    name = "nugget_app_weaver_weaver_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_weaver_weaver_proto",
    ],
)

cc_proto_library(
    name = "nugget_app_avb_avb_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_avb_avb_proto",
    ],
)

cc_proto_library(
    name = "nugget_protobuf_options_cc_proto",
    visibility = ["//visibility:public"],
    deps = [
        "nugget_protobuf_options_proto",
    ],
)

################################################################################
# rpc proto cc_libraries
################################################################################

cc_library(
    name = "keymaster_client_proto",
    srcs = [
        "nugget/app/keymaster/Keymaster.client.cpp",
    ],
    hdrs = [
        "nugget/app/keymaster/Keymaster.client.h",
    ],
    includes = [
        "./nugget/app/keymaster",
    ],
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_keymaster_keymaster_cc_proto",
        "nugget_protobuf_options_cc_proto",
        "@nugget_host_generic_libnos//:libnos",
    ],
)

cc_library(
    name = "weaver_client_proto",
    srcs = [
        "nugget/app/weaver/Weaver.client.cpp",
    ],
    hdrs = [
        "nugget/app/weaver/Weaver.client.h",
    ],
    includes = [
        "./nugget/app/weaver/",
    ],
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_weaver_weaver_cc_proto",
        "nugget_protobuf_options_cc_proto",
        "@nugget_host_generic_libnos//:libnos",
    ],
)

cc_library(
    name = "avb_client_proto",
    srcs = [
        "nugget/app/avb/Avb.client.cpp",
    ],
    hdrs = [
        "nugget/app/avb/Avb.client.h",
    ],
    includes = [
        "./nugget/app/avb/",
    ],
    visibility = ["//visibility:public"],
    deps = [
        "nugget_app_avb_avb_cc_proto",
        "nugget_protobuf_options_cc_proto",
        "@nugget_host_generic_libnos//:libnos",
    ],
)

################################################################################
# rpc proto genrules.
################################################################################

GEN_SERVICE = "$(location @com_google_protobuf//:protoc) --plugin=protoc-gen-nos-client-cpp=$(location @nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp)"

PROTO_ROOT = "$$(dirname $$(dirname $$(dirname $(location nugget/protobuf/options.proto))))"

genrule(
    name = "avb_proto_client_cc_genrule",
    srcs = [
        "nugget/app/avb/avb.proto",
        "nugget/protobuf/options.proto",
        "google/protobuf/descriptor.proto",
    ],
    outs = [
        "nugget/app/avb/Avb.client.cpp",
    ],
    cmd = GEN_SERVICE + " --proto_path=" + PROTO_ROOT +
          " --nos-client-cpp_out=source:$$(dirname $(location nugget/app/avb/Avb.client.cpp)) " +
          "$(location nugget/app/avb/avb.proto)",
    tools = [
        "@com_google_protobuf//:protoc",
        "@nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp",
    ],
)

genrule(
    name = "avb_proto_client_h_genrule",
    srcs = [
        "nugget/app/avb/avb.proto",
        "nugget/protobuf/options.proto",
        "google/protobuf/descriptor.proto",
    ],
    outs = [
        "nugget/app/avb/Avb.client.h",
    ],
    cmd = GEN_SERVICE + " --proto_path=" + PROTO_ROOT +
          " --nos-client-cpp_out=header:$$(dirname $(location nugget/app/avb/Avb.client.h)) " +
          "$(location nugget/app/avb/avb.proto)",
    tools = [
        "@com_google_protobuf//:protoc",
        "@nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp",
    ],
)

genrule(
    name = "weaver_proto_client_cc_genrule",
    srcs = [
        "nugget/app/weaver/weaver.proto",
        "nugget/protobuf/options.proto",
        "google/protobuf/descriptor.proto",
    ],
    outs = [
        "nugget/app/weaver/Weaver.client.cpp",
    ],
    cmd = GEN_SERVICE + " --proto_path=" + PROTO_ROOT +
          " --nos-client-cpp_out=source:$$(dirname $(location nugget/app/weaver/Weaver.client.cpp)) " +
          "$(location nugget/app/weaver/weaver.proto)",
    tools = [
        "@com_google_protobuf//:protoc",
        "@nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp",
    ],
)

genrule(
    name = "weaver_proto_client_h_genrule",
    srcs = [
        "nugget/app/weaver/weaver.proto",
        "nugget/protobuf/options.proto",
        "google/protobuf/descriptor.proto",
    ],
    outs = [
        "nugget/app/weaver/Weaver.client.h",
    ],
    cmd = GEN_SERVICE + " --proto_path=" + PROTO_ROOT +
          " --nos-client-cpp_out=header:$$(dirname $(location nugget/app/weaver/Weaver.client.h)) " +
          "$(location nugget/app/weaver/weaver.proto)",
    tools = [
        "@com_google_protobuf//:protoc",
        "@nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp",
    ],
)

genrule(
    name = "keymaster_proto_client_cc_genrule",
    srcs = [
        "nugget/app/keymaster/keymaster.proto",
        "nugget/app/keymaster/keymaster_defs.proto",
        "nugget/app/keymaster/keymaster_types.proto",
        "nugget/protobuf/options.proto",
        "google/protobuf/descriptor.proto",
    ],
    outs = [
        "nugget/app/keymaster/Keymaster.client.cpp",
    ],
    cmd = GEN_SERVICE + " --proto_path=" + PROTO_ROOT +
          " --nos-client-cpp_out=source:$$(dirname $(location nugget/app/keymaster/Keymaster.client.cpp)) " +
          "$(location nugget/app/keymaster/keymaster.proto)",
    tools = [
        "@com_google_protobuf//:protoc",
        "@nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp",
    ],
)

genrule(
    name = "keymaster_proto_client_h_genrule",
    srcs = [
        "nugget/app/keymaster/keymaster.proto",
        "nugget/app/keymaster/keymaster_defs.proto",
        "nugget/app/keymaster/keymaster_types.proto",
        "nugget/protobuf/options.proto",
        "google/protobuf/descriptor.proto",
    ],
    outs = [
        "nugget/app/keymaster/Keymaster.client.h",
    ],
    cmd = GEN_SERVICE + " --proto_path=" + PROTO_ROOT +
          " --nos-client-cpp_out=header:$$(dirname $(location nugget/app/keymaster/Keymaster.client.h)) " +
          "$(location nugget/app/keymaster/keymaster.proto)",
    tools = [
        "@com_google_protobuf//:protoc",
        "@nugget_host_generic_libnos_generator//:protoc_gen_nos_client_cpp",
    ],
)

################################################################################
# proto libraries
################################################################################

proto_library(
    name = "google_protobuf_descriptor_proto",
    srcs = [
        "google/protobuf/descriptor.proto",
    ],
)

proto_library(
    name = "nugget_app_avb_avb_proto",
    srcs = [
        "nugget/app/avb/avb.proto",
    ],
    deps = [
        "nugget_protobuf_options_proto",
    ],
)

proto_library(
    name = "nugget_app_keymaster_keymaster_proto",
    srcs = [
        "nugget/app/keymaster/keymaster.proto",
        "nugget/app/keymaster/keymaster_defs.proto",
        "nugget/app/keymaster/keymaster_types.proto",
    ],
    deps = [
        "nugget_protobuf_options_proto",
    ],
)

proto_library(
    name = "nugget_app_protoapi_control_proto",
    srcs = [
        "nugget/app/protoapi/control.proto",
    ],
    deps = [
        "nugget_app_protoapi_header_proto",
    ],
)

proto_library(
    name = "nugget_app_protoapi_diagnostics_api_proto",
    srcs = [
        "nugget/app/protoapi/diagnostics_api.proto",
    ],
    deps = [
        "nugget_app_protoapi_header_proto",
    ],
)

proto_library(
    name = "nugget_app_protoapi_gchips_types_proto",
    srcs = [
        "nugget/app/protoapi/gchips_types.proto",
    ],
)

proto_library(
    name = "nugget_app_protoapi_header_proto",
    srcs = [
        "nugget/app/protoapi/header.proto",
    ],
)

proto_library(
    name = "nugget_app_protoapi_testing_api_proto",
    srcs = [
        "nugget/app/protoapi/testing_api.proto",
    ],
    deps = [
        "nugget_app_protoapi_header_proto",
    ],
)

proto_library(
    name = "nugget_app_weaver_weaver_proto",
    srcs = [
        "nugget/app/weaver/weaver.proto",
    ],
    deps = [
        "nugget_protobuf_options_proto",
    ],
)

proto_library(
    name = "nugget_protobuf_options_proto",
    srcs = [
        "nugget/protobuf/options.proto",
    ],
    deps = [
        "@com_google_protobuf//:descriptor_proto",
    ],
)
