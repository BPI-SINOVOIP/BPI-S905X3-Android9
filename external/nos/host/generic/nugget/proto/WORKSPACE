workspace(name = "nugget_host_generic_nugget_proto")
