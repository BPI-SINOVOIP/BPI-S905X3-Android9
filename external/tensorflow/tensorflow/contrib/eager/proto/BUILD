licenses(["notice"])  # Apache 2.0

exports_files(["LICENSE"])

load("//tensorflow/core:platform/default/build_config.bzl", "tf_proto_library")

filegroup(
    name = "all_files",
    srcs = glob(
        ["**/*"],
        exclude = [
            "**/OWNERS",
        ],
    ),
    visibility = ["//tensorflow:__subpackages__"],
)

tf_proto_library(
    name = "checkpointable_object_graph_proto",
    srcs = [
        "checkpointable_object_graph.proto",
    ],
    visibility = ["//tensorflow/contrib/eager/python:__subpackages__"],
)
