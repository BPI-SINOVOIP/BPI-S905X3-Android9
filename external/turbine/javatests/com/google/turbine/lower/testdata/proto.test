=== test/Test.java ===
package test;

// yikes
public class Test<T extends Test.Inner<T>> {
  public interface Inner<T extends Inner<T>> extends Comparable<T> {}
}
