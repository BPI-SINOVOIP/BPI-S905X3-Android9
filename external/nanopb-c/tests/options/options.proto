/* Test nanopb option parsing.
 * options.expected lists the patterns that are searched for in the output.
 */

import "nanopb.proto";

// File level options
option (nanopb_fileopt).max_size = 20;

message Message1
{
    required string filesize = 1;
}

// Message level options
message Message2
{
    option (nanopb_msgopt).max_size = 30;
    required string msgsize = 1;
}

// Field level options
message Message3
{
    required string fieldsize = 1 [(nanopb).max_size = 40];
}

// Forced callback field
message Message4
{
    required int32 int32_callback = 1 [(nanopb).type = FT_CALLBACK];
}

// Short enum names
enum Enum1
{
    option (nanopb_enumopt).long_names = false;
    EnumValue1 = 1;
    EnumValue2 = 2;
}

message EnumTest
{
    required Enum1 field = 1 [default = EnumValue2];
}

// Short enum names inside message
message Message5
{
    enum Enum2
    {
       option (nanopb_enumopt).long_names = false;
       EnumValue1 = 1;
    }
    required Enum2 field = 1 [default = EnumValue1];
}

// Packed structure
message my_packed_struct
{
    option (nanopb_msgopt).packed_struct = true;
    optional int32 myfield = 1;
}

// Message with ignored field
// Note: doesn't really test if the field is missing in the output,
// but atleast tests that the output compiles.
message Message6
{
    required int32 field1 = 1;
    optional int32 field2 = 2 [(nanopb).type = FT_IGNORE];
}

