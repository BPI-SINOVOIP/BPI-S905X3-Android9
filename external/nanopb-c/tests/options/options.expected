char filesize\[20\];
char msgsize\[30\];
char fieldsize\[40\];
pb_callback_t int32_callback;
\sEnumValue1 = 1
Message5_EnumValue1
} pb_packed my_packed_struct;
